### PR TITLE
ergodox_ez / dvorak_42_key layout: convert macros to new format

### PR DESCRIPTION
**Pull request by lucwastiaux**
## Description

Changes to ergodox_ez dvorak_42_key layout.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c [hunk 1]
```diff
@@ -637,25 +590,98 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
...
+            case WINDOWS10_WORKSPACE_RIGHT:
+                SEND_STRING(SS_LGUI(SS_LCTRL(SS_TAP(X_RIGHT))));
+                break;
+            case WINDOWS10_TASK_VIEW:
+                SEND_STRING(SS_LGUI(SS_TAP(X_TAB)));
+                break;
+        // linux screen shortcuts
+        case SCREEN_TAB_LEFT:
+            SEND_STRING(SS_LCTL("a"));
+            SEND_STRING("p");
```
Comment (fauxpark) [inline keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c:616]: These can all be converted to eg.
```suggestion
            tap_code16(C(KC_A));
            tap_code(KC_P);
```

File: keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c [hunk 2]
```diff
@@ -447,44 +457,7 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
```
Comment (fauxpark) [inline keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c:457]: Would you be able to completely get rid of this function?

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c (hunk 2 above)
Comment (lucwastiaux) [reply keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c:457]: yes I can, this was the recommended approach for macros some years back, but I can convert those to the modern way.

Comment (fauxpark) [reply keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c:457]: Yep, I'm trying to remove `fn_actions` and `action_get_macro` as things like direct keycodes (eg. `MO()`) and `process_record_user()` with custom keycodes have basically superseded them.

Comment (lucwastiaux) [reply keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c]: I've removed `action_get_macro` function

File: keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c (hunk 1 above)
Comment (lucwastiaux) [reply keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c:573]: I've started converting new macros to use this format. I will progressively convert all of them but this requires testing. Is this a blocker for this merge ?

File: keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c [hunk 3]
```diff
@@ -637,27 +547,146 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
...
-            SEND_STRING(SS_LGUI(SS_LCTRL(SS_TAP(X_RIGHT))));
+        case SCREEN_2:
+            SEND_STRING(SS_LCTL("a"));
+            SEND_STRING("2");
             break;
-        case WINDOWS10_TASK_VIEW:
-            SEND_STRING(SS_LGUI(SS_TAP(X_TAB)));
+        case SCREEN_3:
+            SEND_STRING(SS_LCTL("a"));
+            SEND_STRING("3");
```
Comment (fauxpark) [inline keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c:609]:
```suggestion
            SEND_STRING(SS_LCTL("a") "3");
```
etc.

File: keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c [hunk 4]
```diff
@@ -637,27 +547,146 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
...
+		case SCREEN_DOWN_JUMP:
+            tap_code(KC_5);
+            tap_code(KC_DOWN);
             break;
 
+        case SCREEN_READREG_1:
+            tap_code16(C(KC_A));
+            tap_code16(S(KC_SCOLON));
+            SEND_STRING("readreg 1");
+            tap_code(KC_ENTER);
```
Comment (fauxpark) [inline keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c:657]:
```suggestion
            SEND_STRING("readreg 1\n");
```
etc.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c (hunk 3 above)
Comment (lucwastiaux) [reply keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c]: done

File: keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c (hunk 4 above)
Comment (lucwastiaux) [reply keyboards/ergodox_ez/keymaps/dvorak_42_key/keymap.c]: done

**Comment by lucwastiaux** [comment]
@fauxpark are you Ok to merge this ?

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**